crates/deltalake-core/src/table/builder.rs

```diff
@@ -150,31 +146,48 @@ pub struct DeltaTableBuilder {
 
 impl DeltaTableBuilder {
     /// Creates `DeltaTableBuilder` from table uri
+    ///
+    /// Can panic on an invalid URI
+    ///
+    /// ```rust
+    /// # use deltalake_core::table::builder::*;
+    /// let builder = DeltaTableBuilder::from_uri("../deltalake-test/tests/data/delta-0.8.0");
+    /// assert!(true);
+    /// ```
     pub fn from_uri(table_uri: impl AsRef<str>) -> Self {
-        Self {
-            options: DeltaTableLoadOptions::new(table_uri.as_ref()),
-            storage_options: None,
-            allow_http: None,
-        }
+        let url = ensure_table_uri(&table_uri).expect("The specified table_uri is not valid");
+        DeltaTableBuilder::from_valid_uri(url).expect("Failed to create valid builder")
     }
 
     /// Creates `DeltaTableBuilder` from verified table uri.
-    /// Will fail fast if specified `table_uri` is a local path but doesn't exist.
+    ///
+    /// ```rust
+    /// # use deltalake_core::table::builder::*;
+    /// let builder = DeltaTableBuilder::from_valid_uri("/tmp");
+    /// assert!(builder.is_ok(), "Builder failed with {builder:?}");
+    /// ```
     pub fn from_valid_uri(table_uri: impl AsRef<str>) -> DeltaResult<Self> {
-        let table_uri = table_uri.as_ref();
-
-        if let UriType::LocalPath(path) = resolve_uri_type(table_uri)? {
-            if !path.exists() {
-                let msg = format!(
-                    "Local path \"{}\" does not exist or you don't have access!",
-                    table_uri
-                );
-                return Err(DeltaTableError::InvalidTableLocation(msg));
+        if let Ok(url) = Url::parse(table_uri.as_ref()) {
+            if url.scheme() == "file" {
+                let path = url.to_file_path().map_err(|_| {
+                    DeltaTableError::InvalidTableLocation(table_uri.as_ref().to_string())
+                })?;
+                ensure_file_location_exists(path)?;
             }
+        } else {
+            ensure_file_location_exists(PathBuf::from(table_uri.as_ref()))?;
         }
 
-        Ok(DeltaTableBuilder::from_uri(table_uri))
+        let url = ensure_table_uri(&table_uri).expect("The specified table_uri is not valid");
```

Comment:
This seems like unnecessarily check if you do `from_uri` already

Reply:
The module level `ensure_table_uri` does some behaviors that needed to be preserved for file URLs in particular. `DeltaTableBuilder::from_uri` and `from_valid_uri` both have slightly different functionality which I do think we might want to get rid of before 1.0